ambari-server/src/main/java/org/apache/ambari/server/actionmanager/ActionDBAccessorImpl.java

```diff
@@ -168,7 +172,18 @@ public ActionDBAccessorImpl(@Named("executionCommandCacheSize") long cacheLimit,
 
   @Inject
   void init() {
-    requestId = stageDAO.getLastRequestId();
+    if (jpaInitialized.get()) {
+      LOG.info("Initializing last request ID from DB ...");
+      requestId = stageDAO.getLastRequestId();
+    }
+  }
+
+  @Subscribe
+  public void jpaInitialized(JpaInitializedEvent event) {
+    LOG.info("JPA initialized event received: {}", event);
+    jpaInitialized.getAndSet(true);
+    init();
+    LOG.info("Set latest request ID.");
```

Comment:
* Shouldn't the call to `init()` be inside an `if (jpaInitialized.getAndSet(true))` block to make sure it's executed only once?
* Log message about request ID belongs to `init()`.

Reply:
Ok.